app/component/itinerary/navigator/NaviUtils.js

```diff
@@ -153,26 +153,40 @@ export const getItineraryAlerts = (
 ) => {
   const alerts = [];
   const canceled = realTimeLegs.filter(leg => leg.realtimeState === 'CANCELED');
-  const legAlerts =
-    realTimeLegs.flatMap(leg => {
-      return leg.alerts?.filter(alert => {
-        const { first } = getFirstLastLegs(realTimeLegs);
-        const startTime = legTime(first.start) / 1000;
-        // show only alerts that are active when
-        // the journey starts
-        if (startTime < alert.effectiveStartDate) {
-          return false;
-        }
-        return (
-          alert.alertSeverityLevel === 'WARNING' ||
-          alert.alertSeverityLevel === 'SEVERE'
-        );
+  let content;
+  const legAlerts = realTimeLegs.flatMap(leg => {
+    return leg.alerts?.filter(alert => {
+      const { first } = getFirstLastLegs(realTimeLegs);
+      const startTime = legTime(first.start) / 1000;
+      // show only alerts that are active when
+      // the journey starts
+      if (startTime < alert.effectiveStartDate) {
+        return false;
+      }
+      return (
+        alert.alertSeverityLevel === 'WARNING' ||
+        alert.alertSeverityLevel === 'SEVERE'
+      );
+    });
+  });
+
+  legAlerts.forEach(alert => {
+    content = (
+      <div className="navi-alert-content">
+        <span className="header"> {alert.alertHeaderText}</span>
+      </div>
+    );
+    if (!messages.get(alert.id)) {
```

Comment:
Sorry for unclear review comment - I meant that filter condition` if (!messages.get(alert.id)) ` could be in the same place where all other filter conditions are checked.